src/tools/wasm-link.cc

```diff
@@ -305,29 +305,52 @@ static void write_table_section(Context* ctx,
 }
 
 static void write_export_section(Context* ctx) {
+  bool seenMemory = false, singleMemoryExport = false;
+
   Index total_exports = 0;
   for (const std::unique_ptr<LinkerInputBinary>& binary: ctx->inputs) {
-    total_exports += binary->exports.size();
+    for (const Export& export_ : binary->exports) {
+      if (export_.kind == ExternalKind::Memory) {
+        if (string_slice_to_string(export_.name) != "memory") {
```

Comment:
Could use `string_slice_eq_cstr` instead. Not a big deal because I should be getting rid of the old `StringSlice` at some point anyway.